pkg/services/live/features/dashboard.go

```diff
@@ -28,16 +45,97 @@ func (h *DashboardHandler) GetHandlerForPath(path string) (models.ChannelHandler
 }
 
 // OnSubscribe for now allows anyone to subscribe to any dashboard
-func (h *DashboardHandler) OnSubscribe(ctx context.Context, _ *models.SignedInUser, e models.SubscribeEvent) (models.SubscribeReply, backend.SubscribeStreamStatus, error) {
-	return models.SubscribeReply{
-		Presence:  true,
-		JoinLeave: true,
-	}, backend.SubscribeStreamStatusOK, nil
+func (h *DashboardHandler) OnSubscribe(ctx context.Context, user *models.SignedInUser, e models.SubscribeEvent) (models.SubscribeReply, backend.SubscribeStreamStatus, error) {
+	parts := strings.Split(e.Path, "/")
+	if parts[0] == "gitops" {
+		// gitops gets all changes for everything, so lets make sure it is an admin user
+		if !user.HasRole(models.ROLE_ADMIN) {
+			return models.SubscribeReply{}, backend.SubscribeStreamStatusPermissionDenied, nil
+		}
+		return models.SubscribeReply{
+			Presence:  true,
+			JoinLeave: true, // ?? likely not necessary
+		}, backend.SubscribeStreamStatusOK, nil
+
+	}
+
+	// make sure can view this dashboard
+	if len(parts) == 2 && parts[0] == "uid" {
```

Comment:
Maybe we need sth more friendly for things like this in the future. Though I have no concrete idea at the moment, but the word `pattern-matching` is around.